src/redux/Checkout/actions.js

```diff
@@ -417,30 +490,34 @@ export function removeItem(item) {
   }
 }
 
-export function validate() {
+export function validate(cart) {
 
   return (dispatch, getState) => {
+    const { shippingAddress } = cart
+    const { address } = getState().checkout
+
     replaceListeners(() => {
-      fetchValidation(dispatch, getState)
+      fetchValidation(dispatch, getState, cart)
     })
-    dispatch(syncAddress())
+
+      dispatch(syncAddress(cart, shippingAddress ?? address))
   }
 }
 
-const _setAddress = createAction(SET_ADDRESS)
+// TODO: Create a specific action
+export const _setAddress = createAction(SET_ADDRESS)
```

Comment:
This was not supposed to be exposed. Maybe better implement changes in `setAddress`.